projects/plugins/jetpack/_inc/client/recommendations/timer/index.jsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import React, { useState, useEffect, useCallback, useRef, useMemo } from 'react';
+
+const Timer = ( { className, expiryDate } ) => {
+	const date = useMemo( () => new Date( expiryDate ), [ expiryDate ] );
+	const [ duration, setDuration ] = useState( date - Date.now() );
+	const intervalRef = useRef();
+
+	const tick = useCallback( () => {
+		setDuration( date - Date.now() );
+	}, [ date, setDuration ] );
+
+	const format = useCallback( ts => {
+		const d = ts / 1000;
+		const days = Math.floor( d / ( 3600 * 24 ) );
+		const hours = Math.floor( ( d - days * 3600 * 24 ) / 3600 );
+		const minutes = Math.floor( ( d - days * 3600 * 24 - hours * 3600 ) / 60 );
+		const seconds = Math.floor( d - days * 3600 * 24 - hours * 3600 - minutes * 60 );
+		const parts = [];
+
+		if ( days > 0 ) {
+			// translators: %d is the number of days, d an abbreviation for days (e.g. 7d). Only translate the latter.
+			parts.push( sprintf( __( '%dd', 'jetpack' ), days ) );
+		}
+
+		if ( hours > 0 ) {
+			// translators: %d is the number of hours, h an abbreviation for hours (e.g. 23h). Only translate the latter.
+			parts.push( sprintf( __( '%dh', 'jetpack' ), hours ) );
+		}
+
+		if ( minutes > 0 ) {
+			// translators: %d is the number of minutes, m an abbreviation for minutes (e.g. 59m). Only translate the latter.
+			parts.push( sprintf( __( '%dm', 'jetpack' ), minutes ) );
+		}
+
+		// translators: %d is the number of seconds, s an abbreviation for seconds (e.g. 59s). Only translate the latter.
+		parts.push( sprintf( __( '%ds', 'jetpack' ), seconds ) );
+
+		return parts.join( ' ' );
+	}, [] );
+
+	useEffect( () => {
+		const id = setInterval( tick, 500 );
+
+		intervalRef.current = id;
+
+		return () => {
+			clearInterval( intervalRef.current );
+		};
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [] );
+
+	return <span className={ className }>{ format( duration ) }</span>;
```

Comment:
If the current time is past `expiryDate` and `duration` becomes negative - should the output of the timer change? Maybe show an "Ended" message in that case?

Reply:
That's probably safer indeed 👍

Reply:
<img width="402" alt="Screen Shot 2022-03-24 at 4 52 42 PM" src="https://user-images.githubusercontent.com/1620183/160016424-95114204-652c-4432-ac11-38814d09173b.png">